src/main/java/frc/robot/common/ControlPanelColorSensor.java

```diff
@@ -0,0 +1,76 @@
+package frc.robot.common;
+
+import com.revrobotics.ColorMatch;
+import com.revrobotics.ColorSensorV3;
+
+import edu.wpi.first.wpilibj.I2C.Port;
+import edu.wpi.first.wpilibj.util.Color;
+
+public class ControlPanelColorSensor {
+
+    public static enum ControlPanelColor {
+        RED(new Color(0.325, 0.453, 0.221)),
+        GREEN(new Color(0.269, 0.550, 0.181)),
+        BLUE(new Color(0.175, 0.469, 0.356)),
+        YELLOW(new Color(0.188, 0.503, 0.310));
+
+        private Color color;
+
+        private ControlPanelColor(Color color) {
+            this.color = color;
+        }
+
+        public Color getColor() {
+            return color;
+        }
+
+        public static ControlPanelColor getColorFromCode(String colorKey) {
+            if (colorKey.equals("R")) {
```

Comment:
You may wish to use a hash map or similar data structure to make this logic simpler.